created a website with recipes, cards, filters, sorts. added css and js.

AgnesSj01 opened the pull request.

AgnesSj01 committed:
created a website with recipes, cards, filters, sorts. added css and js.

AgnesSj01 committed:
added filtering, sorting and random recipie feature

AgnesSj01 committed:
random button, and tidied up the code

AgnesSj01 commented:
Added random recipe button and tidied code

AgnesSj01 committed:
changed the buttons to dropdowns menu, added API, added a button that shows all recipes after clicking the random button, fixed the CSS styling, added a search function and implemented localStorage

AgnesSj01 commented:
Changed the buttons to dropdown menus, added an API, added a button that shows all recipes after clicking the random button, fixed the CSS styling, added a search function, and implemented localStorage.

AgnesSj01 committed:
fixed headern, fixed comments

AgnesSj01 commented:
Netilfy: https://inspiring-sundae-2d353c.netlify.app/

AgnesSj01 committed:
fixed comments and border-radius in the header

AgnesSj01 committed:
Fixed readme

LeonEkelund commented on style.css:
```diff
@@ -0,0 +1,447 @@
+* {
+  box-sizing: border-box;
+}
+body {
+  font-family: "Montserrat", Arial, sans-serif;
+  margin: 0;
+  padding-bottom: 30px;
+  padding-top: 240px;
+}
```
Maybe it would be good to add a overflow x hidden here so that you cant scroll horizontally

LeonEkelund commented on index.html:
```diff
@@ -0,0 +1,83 @@
+  </head>
+  <body>
+    <header class="top-nav">
+      <img src="images/aprikos.jpg" alt="aprikos" class="food" />
```
You can see half of this image (and a bit of the searchbar) at the top when scrolling on iphone 5

LeonEkelund reviewed (commented):
Great job, I just found a few things :)

JennieDalgren commented on script.js:
```diff
@@ -0,0 +1,340 @@
+};
+
+//===================== MAP API RECIPE TO INTERNAL SHAPE =======================
+const mapApiRecipe = (r) => ({
+  // Title with fallback
+  title: r.title || "Untitled",
+  cuisine: (r.cuisines?.[0] || "unknown").toLowerCase(),
+  diet: (r.diets?.[0] || "none").toLowerCase(),
+  popularity: Math.max(
+    1,
+    Math.min(5, Math.round((r.aggregateLikes || 0) / 100))
+  ),
+  time: r.readyInMinutes || 0,
+  img: r.image || "https://via.placeholder.com/600x400?text=No+image",
+  ingredients: (r.extendedIngredients || []).map(
+    (i) => i.original || i.name || ""
+  ),
+});
```
nice function!

JennieDalgren commented on script.js:
```diff
@@ -0,0 +1,340 @@
+//===================== FETCH RECIPES (WITH SIMPLE CACHE) ======================
+const fetchRecipes = async () => {
+  cardsEl.innerHTML = "<p>Fetching recipes…</p>";
+
+  // 1) Try cache first
+  const cached = localStorage.getItem(CACHE_KEY);
+  if (cached) {
+    const saved = JSON.parse(cached);
+    const tooOld = Date.now() - saved.timestamp > CACHE_TIME;
+
+    // Use cached recipes and render immediately
+    if (!tooOld) {
+      cardsEl.innerHTML = "<p>Showing recipes from cache</p>";
+      recipes.length = 0;
+      recipes.push(...saved.items);
+      renderResult();
+      return; // stop here if cache is fresh
+    }
+  }
```
great structure, comments and implementation of function

JennieDalgren reviewed (approved):
Great job with this project!

You have implemented the requirements and almost enough of the stretch goals to recieve a VG.
If you aim for that, implement a loading state when you fetch the data.

Let me know if you will add that, for now I will give you Approved with G

AgnesSj01 committed:
Added loading spinner and improved caching

AgnesSj01 commented:
> Great job with this project!
> 
> You have implemented the requirements and almost enough of the stretch goals to recieve a VG. If you aim for that, implement a loading state when you fetch the data.
> 
> Let me know if you will add that, for now I will give you Approved with G

Hi Jennie! Thanks for the feedback! I decided to go for it and implement a loading state. :)

AgnesSj01 commented:
Hi Jennie! Thanks for the feedback! I decided to go for it and implement a loading state. 🙂

King regards
Agnes Sjösten

________________________________
Från: Jennie Dalgren ***@***.***>
Skickat: den 20 oktober 2025 16:12
Till: Technigo/js-project-recipe-library ***@***.***>
Kopia: AgnesSj01 ***@***.***>; Author ***@***.***>
Ämne: Re: [Technigo/js-project-recipe-library] created a website with recipes, cards, filters, sorts. added css and js. (PR #43)


@JennieDalgren approved this pull request.

Great job with this project!

You have implemented the requirements and almost enough of the stretch goals to recieve a VG.
If you aim for that, implement a loading state when you fetch the data.

Let me know if you will add that, for now I will give you Approved with G

________________________________

In script.js<https://github.com/Technigo/js-project-recipe-library/pull/43#discussion_r2444968450>:

> +const mapApiRecipe = (r) => ({
+  // Title with fallback
+  title: r.title || "Untitled",
+  cuisine: (r.cuisines?.[0] || "unknown").toLowerCase(),
+  diet: (r.diets?.[0] || "none").toLowerCase(),
+  popularity: Math.max(
+    1,
+    Math.min(5, Math.round((r.aggregateLikes || 0) / 100))
+  ),
+  time: r.readyInMinutes || 0,
+  img: r.image || "https://via.placeholder.com/600x400?text=No+image",
+  ingredients: (r.extendedIngredients || []).map(
+    (i) => i.original || i.name || ""
+  ),
+});


nice function!

________________________________

In script.js<https://github.com/Technigo/js-project-recipe-library/pull/43#discussion_r2445139709>:

> +const fetchRecipes = async () => {
+  cardsEl.innerHTML = "<p>Fetching recipes…</p>";
+
+  // 1) Try cache first
+  const cached = localStorage.getItem(CACHE_KEY);
+  if (cached) {
+    const saved = JSON.parse(cached);
+    const tooOld = Date.now() - saved.timestamp > CACHE_TIME;
+
+    // Use cached recipes and render immediately
+    if (!tooOld) {
+      cardsEl.innerHTML = "<p>Showing recipes from cache</p>";
+      recipes.length = 0;
+      recipes.push(...saved.items);
+      renderResult();
+      return; // stop here if cache is fresh
+    }
+  }


great structure, comments and implementation of function

—
Reply to this email directly, view it on GitHub<https://github.com/Technigo/js-project-recipe-library/pull/43#pullrequestreview-3356405442>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BMAXIAQS4P4VS7Y5QDOKA4L3YTUTLAVCNFSM6AAAAACHSMTJA6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTGNJWGQYDKNBUGI>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

JennieDalgren reviewed (approved):
🎊

AgnesSj01 committed:
Update README with app description

Added a brief description of the recipe library app.

AgnesSj01 committed:
Enhance README with project details and objectives

Updated the README to provide a detailed project overview, objectives, features, and technologies used.
